### ISSUE TITLE
Banning a username also bans page names

### ISSUE THREAD

**Issue by hamadax** [PHPfox-Official/phpfox-v4-issues#2642]
# Important

Please follow this template!

----------------------------------------

## What's happened?
Banning a username results in page vanity url being banned as well. Probably happens with any vanity url, groups as well.

## Steps to reproduce:
1. Ban a username
2. Try to make the a page's vanity url that username
3. You get an error.

## What's expected?
If this is by design, a new phrase is needed for this scenario, otherwise, please exclude pages and groups from username bans.

## Browsers and Devices tested
(Example: Chrome on iPhone X, Safari on Macbook, Miscrosoft Edge on Windown 10, Firefox on Ubuntu 16.04, ...)

...

## Server information
(Example: CentOS 7, php 7.1 apache)

...

## phpFox version
(Example: phpFox 4.6.0)

...

## Screenshots
...

**Comment by harrison05** [comment]
Yes, it works as designed. Because when you add Vanity Url for Pages, the page URL will same with user profile URL, so our system will check it with all username or banned username on Site to prevent duplicate.

Regards.

**Comment by hamadax** [comment]
I see. Thanks!

**Comment by hamadax** [comment]
> Yes, it works as designed. Because when you add Vanity Url for Pages, the page URL will same with user profile URL, so our system will check it with all username or banned username on Site to prevent duplicate.

The more I use this feature, the more it makes perfect sense to separate the logic of banning usernames for users and pages. The current logic is restrictive and counterproductive imo. I want to ban usernames "cocacola" with certain wild cards, but not "cocacola" pages. Anyhow, just a suggestion I guess!

**Comment by harrison05** [comment]
I understand your request but with the current source code and workflow, it so difficult to separate them to 2 sections with no relation. 
But we will push this request to our review process and inform you if have any update.

Regards.
